core/src/main/java/org/apache/iceberg/MetadataColumns.java

```diff
@@ -50,7 +50,7 @@ private MetadataColumns() {}
   public static final int SPEC_ID_COLUMN_ID = Integer.MAX_VALUE - 4;
   public static final String SPEC_ID_COLUMN_DOC = "Spec ID used to track the file containing a row";
   public static final NestedField SPEC_ID =
-      NestedField.required(
+      NestedField.optional(
```

Comment:
In Spark 4.0, the metadata columns are nullable, so need to change this field to optional. 
https://github.com/apache/spark/pull/50246#discussion_r2023825620